packages/protocol/contracts/dapis/rrpDapiServer.sol

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+import "../AirnodeRrp.sol";
+import "./CustomReducer.sol";
+
+contract RrpDapiServer is CustomReducer {
+    struct DapiParameters {
+        uint256 minResponsesToReduce;
+        int256 toleranceInPercentages;
+        uint256 requesterIndex;
+        bytes32[] templateIds;
+        address[] designatedWallets;
+        address reduceAddress;
+        bytes4 reduceFunctionId;
+        }
+    
+    AirnodeRrp public airnodeRrp;
+    mapping(uint256 => DapiParameters) private dapiIdToParameters;
+    uint256 public nextDapiID = 1;
+    uint256 public nextDapiRequestIndex = 1;
+    mapping(bytes32 => uint256) private requestIdToDapiRequestIndex;
+    mapping(uint256 => int256[]) private dapiRequestIndexToResponses;
+    mapping(uint256 => uint256) private dapiRequestIndexToDapiId;
+
+    constructor(address _airnodeRrp) {
+        airnodeRrp = AirnodeRrp(_airnodeRrp);
+    }
+
+    function registerDapi
+    (
+        uint256 minResponsesToReduce,
+        int256 toleranceInPercentages,
+        uint256 requesterIndex,
+        bytes32[] calldata templateIds,
+        address[] calldata designatedWallets,
+        address reduceAddress,
+        bytes4 reduceFunctionId
+    )
+        external
+        returns (uint256 dapiId)
+    {
+      dapiIdToParameters[nextDapiID] = DapiParameters(
+        minResponsesToReduce,
+        toleranceInPercentages,
+        requesterIndex,
+        templateIds,
+        designatedWallets,
+        reduceAddress,
+        reduceFunctionId
+      );
+      dapiId = nextDapiID++;
+    }
+
+
+    function makeDapiRequest(uint256 dapiId, uint256 requesterIndex)
```

Comment:
This should include a `bytes parameters` argument that gets used in all `makeRequest`s

Reply:
yea, i think it was empty as a TODO

Reply:
Is it reasonable to assume all templates would use the same parameters?

Reply:
You are asking if it should be `bytes` or `bytes[]` right? Even if different endpoints may require differently formatted parameters I don't think it's feasible to specify parameters for each request. Then, the Airnodes should be expected to accept request-time parameters in the same format (so `bytes`).